tests/apollo/test_skvbc_restart_recovery.py

```diff
@@ -320,9 +318,72 @@ async def test_restarting_f_replicas_for_view_change(self, bft_network):
         await bft_network.wait_for_fast_path_to_be_prevalent(
             run_ops=lambda: skvbc.run_concurrent_ops(num_ops=20, write_weight=1), threshold=20)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: c == 0, rotate_keys=True)
+    @with_constant_load
+    @verify_linearizability()
+    async def test_recovering_fast_path(self, bft_network, skvbc, constant_load, tracker):
+        """
+        The Apollo test, which should be part of the test_skvbc_restart_recovery suite needs to implement the following steps:
+
+        1. Start all Replicas and Introduce constantly running Client requests in the background.
+        2. Verify that the system is processing client requests on Fast Path.
+        3. Restart the Primary to initiate View Change.
+        4. Verify the View Change succeeded.
+        5. Wait for the system to recover Fast commit path again (Here we will have to set a time limit for which we expect the system to recover to Fast path. We can start with 1 minute interval.)
+        6. Stop 1 Non Primary replica to transition the system to Slow path.
+        7. Verify the system is making progress on Slow path.
+        8. Restart the Primary to initiate View Change.
+        9. Start the Replica we stopped in step 6 and verify the View Change succeeded.
+        10. Goto Step 2.
+        """
+        # start replicas
+        [bft_network.start_replica(i) for i in bft_network.all_replicas()]
+
+        loop_count = 0
+        while (loop_count < 1):
+            loop_count = loop_count + 1
+
+            log.log_message("view = await bft_network.get_current_view(")
+            view = await bft_network.get_current_view()
+
+            primary = await bft_network.get_current_primary()
+            bft_network.stop_replica(primary)
+            bft_network.start_replica(primary)
```

Comment:
After the restart we need the metrics server to be started. In many of the runs I performed locally we get a blocking on **_await_replicas_in_state_transfer** waiting to get status of the replica that was restarted and eventually timeouting. One way to solve it is to introduce a waiting period (5 to 10 seconds should be enough I think). We might want to add a note to correct the getting of the metrics to not get stuck if the metric server on the target replica is not live yet and revisit once this is implemented.

Reply:
Done